### DISCUSSION TITLE
Filtering Dataset with PGArray

### DISCUSSION THREAD

**Discussion by i-h8-github** [jeremyevans/sequel#2112] [General]
Hello,

Using http://sequel.jeremyevans.net/rdoc/files/doc/querying_rdoc.html#label-Filters as a reference, I find the following interesting:

```ruby
Artist.where(id: [1, 2]) # yields expected rows
Artist.where(id: Sequel.pg_array([1, 2])) # PG::UndefinedFunction: ERROR:  operator does not exist: integer = integer[]
```

I have a PGArray that I pull from another table to use, and I can call `to_a` on it to get the desired functionality, but I'm curious why I can't use it that way directly?

**Comment by jeremyevans** [comment]
You cannot use it that way directly for IN because treating the PGArray as inclusion would break its current use as equality:

```
# Check whether int_array_column = ARRAY[1, 2]
Artist.where(int_array_column: Sequel.pg_array([1, 2]))
```

**Reply by i-h8-github** [reply]
I see, that makes sense. Thank you.
